src/link.ts

```diff
@@ -298,32 +298,39 @@ export class Link implements esr.AbiProvider {
     ): Promise<TransactResult> {
         const t = transport || this.transport
         const broadcast = options ? options.broadcast !== false : true
-        // eosjs transact compat: upgrade to transaction if args have any header fields
-        let anyArgs = args as any
-        if (
-            args.actions &&
-            (anyArgs.expiration ||
-                anyArgs.ref_block_num ||
-                anyArgs.ref_block_prefix ||
-                anyArgs.max_net_usage_words ||
-                anyArgs.max_cpu_usage_ms ||
-                anyArgs.delay_sec)
-        ) {
-            args = {
-                transaction: {
-                    expiration: '1970-01-01T00:00:00',
-                    ref_block_num: 0,
-                    ref_block_prefix: 0,
-                    max_net_usage_words: 0,
-                    max_cpu_usage_ms: 0,
-                    delay_sec: 0,
-                    ...anyArgs,
-                },
+        try {
```

Comment:
Why is this needed?

Reply:
To catch and throw errors during `this.createRequest` mainly.

Reply:
Added a new commit that moved this into the createRequest call, which I think also addresses your other comment about it potentially being called multiple times?

Reply:
Also just for clarity, it allows our error messages to intercept things like this:

![image](https://user-images.githubusercontent.com/677686/85054374-efbc5100-b169-11ea-9355-83f2f7b9f508.png)

Reply:
That's not a very helpful error message to the user, not showing those errors was intentional, it's more of a "you're calling it wrong" than a "something unexpected went wrong with your transaction"

Reply:
They are not, but I'd rather get screenshots from users with this error message than having to tell them to open their browser console and give me the error reported there.

Reply:
Ok, guess it's not a huge deal now that the app can choose to hide error/success messages.